### ISSUE TITLE
Добавить возможность к каждой конфигурации ограничить доступ пользователям

### ISSUE THREAD

**Issue by Dimanchik00** [BlizD/Tasks#53]
При ведении в базе нескольких проектов (конфигураций), не всем разработчикам нужен доступ ко всем проектам.Поэтому хотелось бы опционально иметь возможность устанавливать доступность конфигураций определенным пользователям.

**Comment by BlizD** [comment]
@Dimanchik00 
Добрый день.
>При ведении в базе нескольких проектов (конфигураций)

Речь идет об отдельных ветках в справочнике "Задачи"?
Вот так
- Проект №1
-- Задача 34
-- Задача 56
- Проект №17
-- Задача 23
-- Задача 35

Необходим отдельный доступ к "Проекту №1" и к "Проекту №2"?

**Comment by Dimanchik00** [comment]
Можно и так. Тогда надо иметь возможность давать доступ к Проект№1 списку пользователей. И не показывать тем кого там нет. Имелся в виду следующий сценарий. У задачи есть реквизит конфигурация. К каждой записи справочника конфигурация давать доступ определенному списку пользователей (участникам проекта). Тогда обязательно заполнение этого реквизита в задачах. Кстати, при создании новой задачи родитель указывается, а конфигурация автоматически из родителя не подставляется.

**Comment by BlizD** [comment]
Понятно, надо подумать как это реализовать.
Не хотелось бы реализовывать через RLS.

**Comment by BlizD** [comment]
@Dimanchik00 
Подскажи, насколько это важно, разделять такую видимость?
Там есть информация, которую нельзя видеть другим пользователям?

**Comment by Dimanchik00** [comment]
Да, если вести проекты по разработке в одной компании в одной базе. Команд может быть несколько и конфиденциальность однозначно нужна.

**Comment by BlizD** [comment]
@Dimanchik00 
>Да, если вести проекты по разработке в одной компании в одной базе. Команд может быть несколько и конфиденциальность однозначно нужна.

А в Вашем случае уже есть такая потребность? или на будущее?

**Comment by Dimanchik00** [comment]
Почти. Первый большой проект туда портируем уже.

**Comment by BlizD** [comment]
@Dimanchik00 
Через RLS не хочу делать по нескольким причинам:
* текущий код не предусмотрен для RLS, будут сыпаться ошибки из разных мест;
* общее замедление системы из за добавления RLS;
* большие трудозатраты;

Как мне кажется если и делать, то через программную реализацию:
* обеспечить видимость в формах списках, форма выбора, форме элемента для задач;
* подправить канбан доску;
* подправить отчет по задачам;

Надо понимать, что в этом случае у пользователя будет возможность через консоль запросов/произвольные отчет получить закрытую информацию для него.

**Comment by JohnyDeath** [comment]
А может добавить справочник "Проекты" и соответствующий реквизит в задачу?
По возможности вытащить его во все места (регистры, документы)
Замедления особо и не заметишь.
Трудозатраты будут связаны только с вытаскиванием проекта во все таблицы. 
Сами запросы не долго делать, особенно если использовать шаблоны.

Но так будет правильно и архитектурно и в дальнейшем не будет приводить к головной боли при развитии функционала.

Мне, возможно, тоже такое понадобится. Есть несколько команд/проектов, которые друг о друге вообще ничего не знают. Более того, будет возможен доступ и заказчику, чтобы, например, посмотреть состояние дел.

**Comment by BlizD** [comment]
@JohnyDeath 
там же есть еще объекты БСП не знаю как они себя поведут в связке с RLS?

**Comment by JohnyDeath** [comment]
@BlizD что например тебя смущает? Какие именно объекты?
Доступ к которым не надо ограничивать и не будут в РЛС участвовать

**Comment by BlizD** [comment]
@JohnyDeath 
При настройке БСП, были определяемые типы, в которых указывал Задачу как тип.
Не знаю участвует ли задача в этом случае в каких то служебных регистрах.

**Comment by BlizD** [comment]
@JohnyDeath 
>А может добавить справочник "Проекты" и соответствующий реквизит в задачу?
По возможности вытащить его во все места (регистры, документы)
Замедления особо и не заметишь.

Может лучше добавить какой то пространный реквизит типа: "Область видимости", "Группа доступа". Не всегда же именно проекты будут разделением.

**Comment by JohnyDeath** [comment]
Да, наверное стоит подумать над "областью видимости", а точнее "доступные проекты" для каждого пользователя. Но по факту разделителем все равно должен остаться Проект, как мне кажется.
В учетных системах такой реквизит - Организация. И некоторые пользователи имеют доступ к нескольким организациям

**Comment by capitoshko** [comment]
Давайте поднимем задачу ещё раз. Я склоняюсь к варианту отдельного справочника Проекты и RLS. Что скажете?

**Comment by BlizD** [comment]
По поводу отдельного справочника "Проекты" - пока все таки против него.
Вместо него необходимо использовать термин "Основная задача". То есть основная задача (проект) и в него входят другие подчиненные задачи. 
Что касается RLS и проверки прав доступа, то в целом не против.

**Comment by capitoshko** [comment]
Использовать рлс на документ это как то не очень

**Comment by BlizD** [comment]
>Использовать рлс на документ это как то не очень
О каком документе говоришь?

**Comment by capitoshko** [comment]
Морі. Є о справляння. Сори
